packages/components/src/Button/Button.tsx

```diff
@@ -54,7 +54,7 @@ interface ButtonLinkProps extends ButtonFoundationProps {
 }
 
 interface BaseActionProps extends ButtonFoundationProps {
-  readonly variation?: "work" | "learning";
+  readonly variation?: "work" | "learning" | "destructive";
```

Comment:
This shouldn't require changes to `Button`. There are currently restrictions on Button that make us not want this option available here.

Reply:
Fair point but without adding this I see no easy way 
![image](https://user-images.githubusercontent.com/84454304/132744684-eca95252-7578-4ff2-9dd1-4f45d20309ed.png)

Reply:
Modal's primaryAction is of type ButtonProp which allows variation to be either 'work | learning' or 'destructive'